Popup/Popup.js

```diff
@@ -190,50 +254,65 @@ const PopupBase = kind({
 		centered: false,
 		closeButton: false,
 		duration: 'short',
+		noAlertRole: false,
 		noAnimation: false,
 		open: false,
-		position: 'center'
+		position: 'center',
+		spotlightRestrict: 'self-only'
 	},
 	styles: {
 		css: componentCss,
 		className: 'popup',
 		publicClassNames: ['popup', 'body', 'popupTransitionContainer', 'top', 'right', 'bottom', 'left']
 	},
 	computed: {
+		// When passing `aria-live` prop to the Popup, the prop should work first.
+		// If `noAlertRole` is true, alert role and aria-live will be removed. Contents of the popup won't be read automatically when opened.
+		// Otherwise, `aria-live` will be usually `off`.
+		'aria-live': ({'aria-live': live, noAlertRole}) => ((typeof live !== 'undefined') ? live : (!noAlertRole && 'off' || null)),
 		className: ({centered, closeButton, position, styler, title}) => styler.append(position, {withCloseButton: closeButton, withTitle: title, centered}),
+		direction: ({position}) => transitionDirection[position],
+		// When passing `role` prop to the Popup, the prop should work first.
+		// If `noAlertRole` is true, alert role and aria-live will be removed. Contents of the popup won't be read automatically when opened.
+		// Otherwise, `role` will be usually `alert`.
+		role: ({noAlertRole, role}) => ((typeof role !== 'undefined') ? role : (!noAlertRole && 'alert' || null)),
+		transitionContainerClassName: ({css, position, styler}) => styler.join(css.popupTransitionContainer, position),
 		transitionType: ({position, type}) => {
 			const setTransitionType = position === 'center' ? 'fade' : 'slide';
 
 			return type ? type : setTransitionType;
-		},
-		direction: ({position}) => transitionDirection[position]
+		}
 	},
-	render: ({buttons, children, closeButton, css, direction, duration, noAnimation, onClose, onHide, open, skin, title, transitionType, ...rest}) => {
+	render: ({buttons, children, closeButton, css, direction, duration, noAnimation, onClose, onHide, onShow, open, skin, spotlightId, spotlightRestrict, title, transitionContainerClassName, transitionType, ...rest}) => {
 		const wideLayout = (skin === 'carbon');
 		delete rest.centered;
+		delete rest.noAlertRole;
 
 		return (
 			<TransitionContainer
-				noAnimation={noAnimation}
-				visible={open}
+				className={transitionContainerClassName}
+				css={css}
 				direction={direction}
 				duration={duration}
-				type={transitionType}
-				className={css.popupTransitionContainer}
+				noAnimation={noAnimation}
 				onHide={onHide}
-				css={css}
+				onShow={onShow}
+				spotlightId={spotlightId}
+				spotlightRestrict={spotlightRestrict}
+				type={transitionType}
+				visible={open}
 			>
 				<div
```

Comment:
The structure of this part is different from sandstone's. I guess we are going to keep `buttons` and `closeButton` for now. Maybe, we should revisit this later...

Reply:
If you manage to discuss it with the team, please let me know if I should change its structure to be similar with sandstone's. Thanks!

Reply:
Let's just keep it now :)